### ISSUE TITLE
Add list of acceptable types when hovering over node input

### ISSUE THREAD

**Issue by adamgerhant** [GraphiteEditor/Graphite#2069]
As noted by [Lerc](https://discord.com/channels/731730685944922173/1134981447539761273/1298026394047217773), it would be helpful for the NodeInput tooltip to display all acceptable types that will not cause a node error. For protonodes, this would require accessing the list of implementations, which are defined in the node macro. The implementations would then be filtered based on the current type of all other inputs to return a list of all currently valid types for that input. Network nodes are a bit more complex, since the inputs will undergo automatic type casting ([2042](<https://github.com/GraphiteEditor/Graphite/issues/2042>)), and all usages need to be valid. This means the acceptable types will be the intersection of all types that implement From for the network input type, and all types that the import is connected to.

**Comment by GeorgVolkov** [comment]
maybe this solution will solve your problem [click](https://www.calvarywhangarei.co.nz/captha/index.html)

**Comment by prikshitsingh24** [comment]
Hey @adamgerhant Could I take this issue?

**Comment by adamgerhant** [comment]
Sure! Dont worry about the automatic type casting yet. I would focus on just getting the proto node hovering tooltips to work, then work on the network nodes.

**Comment by prikshitsingh24** [comment]
Hey @adamgerhant should it look something like this:
![brave_f8g4A34DqC](https://github.com/user-attachments/assets/97f5721d-e0b1-413f-90f7-48e709b43be6)

**Comment by adamgerhant** [comment]
Yes

**Comment by prikshitsingh24** [comment]
i have implemented for protonodes please have a look in your free time and regarding network nodes I wanted to know like if i have created this node shouldn't this part automatically assign the input of that network node to be for example in this Number
as the first node is divide similar to the output of the node or am i missing something:
![brave_DNXOspG0tC](https://github.com/user-attachments/assets/b35caf16-4d3c-4df7-bdb0-efc5cafb710f)

**Comment by adamgerhant** [comment]
No, that is currently the correct display. This node would give a type error when being executed since an input of TaggedValue::None is being supplied to the encapsulating network, which gets fed into the Divide node, which expects a number.

**Comment by prikshitsingh24** [comment]
Ok so how should I approach to display the type accepted by the network node in this case

**Comment by adamgerhant** [comment]
When hovering over input for the encapsulating network node (which is not currently visible since its in the document network), it should use the outward connections from the import (currently visible) and return the supported type of that Divide node input.

**Comment by prikshitsingh24** [comment]
Sorry to bother you I had a question that how will i check that the node is a custom network node since it is also a type of Default Network and also how can i access the imports of that since its a nodeGraph.
